main_window/read_thread.py

```diff
@@ -72,13 +71,12 @@ def run(self):
                 try:
                     parsed_data: Dict[int, any] = self.packet_parser.extract(byte_stream)
                     self.rocket_data.addBundle(parsed_data)
-                    curr_offset = byte_stream.tell()
 
                     # notify UI that new data is available to be displayed
                     self.sig_received.emit()
                 except Exception as e:
-                    LOGGER.exception("Error decoding new data!")  # Automatically grabs and prints exception info
-                    byte_stream.seek(curr_offset+1)
+                    LOGGER.error("Error decoding new data!")
+                    # Just discard rest of data
```

Comment:
Need to get it out of the `while byte_stream.tell() < end:` loop somehow to actually discard it.

Reply:
Added one approach. We should discuss our strategy, including what you mentioned below.